src/app/modules/core/services/chain.service.ts

```diff
@@ -1,13 +1,18 @@
 import { Injectable } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
 
-import { switchMap } from 'rxjs/operators';
+import { switchMap, startWith } from 'rxjs/operators';
 
 import { BeaconNodeService } from './beacon-node.service';
 
 import {
   ValidatorParticipationResponse,
 } from 'src/app/proto/eth/v1alpha1/beacon_chain';
+import { interval } from 'rxjs';
+
+const MILLISECONDS_PER_SLOT = 12000
+const SLOTS_PER_EPOCH = 32
```

Comment:
I think these two should be moved to `constants.ts` because their values will be correct for all components.